### PR TITLE
[FEATURE] Display network unavailable banner when no network connection is available

### PR DESCRIPTION
**Pull request by bkonyi**
Currently set to appear over every screen when network connection is lost. Once the user taps "CONTINUE OFFLINE", the banner will not be shown unless network connection is restored and then lost again. Banner disappears once network connection is restored.

**Screenshots:**
Simulator Screen Shot - iPhone 11 Pro Max - 2020-03-24 at 16 07 44

**---------------------------------------------------------------------------------------------------**

Simulator Screen Shot - iPhone 11 Pro Max - 2020-03-24 at 16 07 24

**---------------------------------------------------------------------------------------------------**

Simulator Screen Shot - iPhone 11 Pro Max - 2020-03-24 at 16 07 01

### CODE REVIEW

File: lib/src/ui/widgets/network_unavailable_banner.dart [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        ],
+      ),
+    );
+  }
+
+  @override
+  _NetworkUnavailableBannerState createState() =>
+      _NetworkUnavailableBannerState();
+
+  final Connectivity connectivity;
```
Comment (ferndot) [inline lib/src/ui/widgets/network_unavailable_banner.dart]: Nit: move this to the top of the class

File: test/ui/widgets/network_connectivity_banner_test.dart [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    Future<void> testDismissBanner(WidgetTester tester) async {
+      // Dismiss the banner.
+      final l10n = await AppLocalizations.delegate.load(Locale('en'));
+      await tester.tap(find.text(l10n.networkUnavailableBannerContinueOffline));
+      await tester.pumpAndSettle();
+      expect(find.byType(MaterialBanner), findsNothing);
+      expect(find.byType(Container), findsOneWidget);
+    }
+
+    testWidgets('and is dismissable', (tester) async {
```
Comment (ferndot) [inline test/ui/widgets/network_connectivity_banner_test.dart]: This is super clean 👍

**Review by ferndot — DISMISSED** [review]
This is great. Eventually, we might want to investigate a way to automatically wrap each screen with this. Perhaps, we could implement a custom scaffold.

**Comment by ferndot** [comment]
Please rebase this branch to remove the conflicts and then it can be merged!

**Review by dnfield — COMMENTED** [review]
@timsneath - is continue offline really an option for MVP?

File: lib/src/ui/widgets/network_unavailable_banner.dart (hunk 1 above)
Comment (bkonyi) [reply lib/src/ui/widgets/network_unavailable_banner.dart]: Done.

File: test/ui/widgets/network_connectivity_banner_test.dart (hunk 2 above)
Comment (bkonyi) [reply test/ui/widgets/network_connectivity_banner_test.dart]: Thanks!

**Comment by bkonyi** [comment]
> This is great. Eventually, we might want to investigate a way to automatically wrap each screen with this. Perhaps, we could implement a custom scaffold.

Yeah, I was thinking of doing a custom scaffold but wasn't sure if that's what we'd want at this point. Shouldn't be hard to do in the future if we decide to take that route.

**Comment by bkonyi** [comment]
Ugh, rebased and got some other commits in here... I think this should still be good to go? @joshua-s

File: lib/src/ui/widgets/network_unavailable_banner.dart [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+                        ),
+                        actions: <Widget>[
+                          FlatButton(
+                            child: Text(
+                              AppLocalizations.of(context)
+                                  .networkUnavailableBannerContinueOffline,
+                            ),
+                            onPressed: () async =>
+                                setState(() => _dismissed = true),
+                          ),
```
Comment (dnfield) [inline lib/src/ui/widgets/network_unavailable_banner.dart]: I don't think we should show this until there's actually offline capabilities, which is not being targetted for v1.

**Review by dnfield — CHANGES_REQUESTED** [review]
It doesn't make sense to let people continue offline if they cannot actually save any of this data anyway.

File: lib/src/ui/widgets/network_unavailable_banner.dart (hunk 3 above)
Comment (bkonyi) [reply lib/src/ui/widgets/network_unavailable_banner.dart]: I've added a `dismissible` property which defaults to false so we can re-enable this button later.

**Comment by bkonyi** [comment]
> It doesn't make sense to let people continue offline if they cannot actually save any of this data anyway.

See the latest commits for the addition of a property which controls the visibility of the "CONTINUE OFFLINE" button.

**Review by dnfield — APPROVED** [review]
LGTM

**Pull request merged**